feat: add STEAMAPPBRANCH build arg for beta branch support

Diagounal opened the pull request:
- Added `STEAMAPPBRANCH` ARG to Dockerfile to allow installing beta branches (e.g. `unstable`).
- Updated `steamcmd` command to conditionally use `-beta` flag.
- Added example usage to `docker-compose.yml`.
- Updated `README.md` with documentation, usage examples, and a note about Build 42 multiplayer.

Diagounal committed:
feat: add STEAMAPPBRANCH build arg for beta branch support

- Added `STEAMAPPBRANCH` ARG to Dockerfile to allow installing beta branches (e.g. `unstable`).
- Updated `steamcmd` command to conditionally use `-beta` flag.
- Added example usage to `docker-compose.yml`.
- Updated `README.md` with documentation, usage examples, and a note about Build 42 multiplayer.

efedleikudl reviewed (approved):
Very nice feature, works well

flan6 commented:
@Diagounal could you add the `STEAMAPPBRANCH` variable to .env.template please?

Diagounal committed:
refactor: Ensure correct Steam branch is used for runtime updates

STEAMAPPBRANCH is now a standard environment variable (ENV) configured via .env.template, replacing the previous ARG. This change ensures the server update logic within entry.sh now correctly uses the specified Steam branch, preventing version inconsistencies.

Darkblader24 commented:
Mh it doesn't seem to work for me. It still installs version 41, even though I use "STEAMAPPBRANCH=unstable" in the .env file

Diagounal committed:
fix: Correctly set STEAMAPPBRANCH via .env using ARG-to-ENV mapping

The build process was failing to receive the STEAMAPPBRANCH value from the .env file. This commit implements the standard ARG-to-ENV mapping pattern to bridge the variable from docker-compose.yml's args section to the Dockerfile's build-time instructions.

This ensures the user's configured Steam branch is consistently used for both the initial game download and subsequent runtime updates.

Diagounal commented:
> Mh it doesn't seem to work for me. It still installs version 41, even though I use "STEAMAPPBRANCH=unstable" in the .env file

Fixed! My apologies for the earlier error.

The problem was that the `STEAMAPPBRANCH` from the `.env` file wasn't being passed to the image build step. I've now pushed the change to correctly map the variable as a build ARG.

Mirkoffcito reviewed (commented):
I've been trying it out for a couple of days with both the early week `unstable` version and the latest `42.13.1` version, both worked great OOB.

Worth noting (and might be an useful add to the README) that you will probably need to rebuild the image if you're trying out a new branch.

MaicomMR commented:
Are you able to build the image successfully? I don't think this is related to the code in this PR, but I'm getting this crash on every build attempt.


```
12.54 Waiting for user info...OK
16.11 ERROR! Failed to install app '380870' (Missing configuration)
16.74 Unloading Steam API...OK
17.09 
------
Dockerfile:33

--------------------

  32 |     # Set the entry point file permissions
  33 | >>> RUN set -x \
  34 | >>>   && mkdir -p "${STEAMAPPDIR}" \
  35 | >>>   && chown -R "${USER}:${USER}" "${STEAMAPPDIR}" \
  36 | >>>   && bash "${STEAMCMDDIR}/steamcmd.sh" +force_install_dir "${STEAMAPPDIR}" \
  37 | >>>   +login anonymous \
  38 | >>>   +app_update "${STEAMAPPID}" ${STEAMAPPBRANCH:+-beta "$STEAMAPPBRANCH"} validate \
  39 | >>>   +quit
  40 |     

--------------------

failed to solve: process "/bin/sh -c set -x   && mkdir -p \"${STEAMAPPDIR}\"   && chown -R \"${USER}:${USER}\" \"${STEAMAPPDIR}\"   && bash \"${STEAMCMDDIR}/steamcmd.sh\" +force_install_dir \"${STEAMAPPDIR}\"   +login anonymous   +app_update \"${STEAMAPPID}\" ${STEAMAPPBRANCH:+-beta \"$STEAMAPPBRANCH\"} validate   +quit" did not complete successfully: exit code: 8
```

Mirkoffcito commented:
@MaicomMR Yup... Every once in a while I'll get an error but probably related to my WiFi and some connection error with the Steam API. Other than that it works smoothly, for me.

Last night after updating to `STEAMAPPBRANCH=42.13.1` I just did: `docker compose up -d --build --force-recreate` and server was up and running, no progress lost.

Danixu commented:
Thanks for the improvements!!! 🙂​. I think that the idea is good and I want to try to improve it to build also the beta versions directly on Dockerhub.
